yt/funcs.py

```diff
@@ -1195,6 +1195,16 @@ def validate_sequence(obj):
         )
 
 
+def validate_field_key(key):
+    if (
+        isinstance(key, tuple)
+        and len(key) == 2
+        and all(isinstance(_, str) for _ in key)
+    ):
+        return
+    raise TypeError(f"Expected a 2-tuple of strings, received invalid field key: {key}")
```

Comment:
Feel free to ignore, but I think the message would be clearer as follows:
```suggestion
    raise TypeError(f"Expected a tuple of (field type, field name), received invalid field key: {key}")
```

Reply:
I think all these bits are valuable, I'll incorporate your suggestion but keep the typing stuff too.

Reply:
actually now I remember why I didn't do it : "field type" is somewhat misleading because sometimes we call this element "particle type", but then if we embrace that generality I'm thinking the message could be more confusing than helping. What do you think ?

Reply:
Hmmh didn't think at this. What about `(field or particle type, field name)`? In any case, this isn't valid python and is expected to be parsed by a human-brain™ instead.

Reply:
alright let's ride with this